katsdptelstate/lua_scripts/get.lua

```diff
@@ -1,7 +1,11 @@
-local result = redis.pcall('GET', KEYS[1])
-if type(result) == 'table' then
+local type_ = redis.call('TYPE', KEYS[1])['ok']
+if type_ == 'none' then
+    return {false, type_}
+elseif type_ == 'string' then
+    return {redis.call('GET', KEYS[1]), type_}
+elseif type_ == 'zset' then
     local last = redis.call('ZREVRANGEBYLEX', KEYS[1], '+', '-', 'LIMIT', 0, 1)
-    return {last[1], true}
+    return {last[1], type_}
 else
-    return {result, false}
+    return {redis.call('HGETALL', KEYS[1]), type_}
```

Comment:
I did it this way so that if the type is none of the above then you get an error from redis that says you're operating on the wrong type of key. If I used an elseif then I'd need to synthesize such an error with Lua code. I'll add a comment to that effect.